hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoTransferHandler.java

```diff
@@ -210,15 +199,25 @@ private void handleHbarTransfers(
     }
 
     private boolean receivesFungibleValue(
-            final AccountID target, final CryptoTransferTransactionBody op) {
+            final AccountID target, final CryptoTransferTransactionBody op, final ReadableAccountStore accountStore) {
         for (var adjust : op.getTransfers().getAccountAmountsList()) {
-            if (adjust.getAmount() > 0 && adjust.getAccountID().equals(target)) {
+            final var unaliasedAccount = accountStore.getAccount(adjust.getAccountID());
```

Comment:
This block L204-209 is duplicated with L215-220, would be nice to extract.

Reply:
yes , will do in the next PR for modularization.